src/flask_restalchemy/resources/querybuilder.py

```diff
@@ -33,21 +33,11 @@ def query_from_request(model, model_serializer, request, query=None):
     if not query:
         query = model.query
 
-    query = config_query_from_request(query, model, model_serializer, request)
-
-    if 'page' in request.args:
-        data = query.paginate()
-        return {
-            'page': data.page,
-            'per_page': data.per_page,
-            'count': data.total,
-            'results': [model_serializer.dump(item) for item in data.items]
-        }
-    else:
-        data = query.all()
-        return [model_serializer.dump(item) for item in data]
+    query = collection_query_builder(query, model, model_serializer, request.args)
+
+    return query
 
-def config_query_from_request(query, model, model_serializer, request):
+def collection_query_builder(parent_query, model_class, model_serializer, args):
```

Comment:
Could you add 'args' to the doc string? Also remove the docs for 'request'.

Reply:
Rename it to `create_collection_query`